backend/apps/recommendations/services/github_analyzer.py

```
@@ -0,0 +1,258 @@
+            return response.json()
+        except Exception as e:
+            logger.error(f"Failed to get user data for {username}: {e}")
+            return None
+    
+    def _get_user_repos(self, username: str) -> List[Dict]:
+        """Get user repositories."""
+        repos = []
+        page = 1
+        while True:
+            try:
+                response = requests.get(
```

Ruff (0.15.20)

[error] Probable use of `requests` call without timeout

(S113)